.editorconfig

```diff
@@ -8,255 +8,596 @@ end_of_line = LF
+#csharp_style_pattern_matching_over_is_with_cast_check = true
 dotnet_diagnostic.IDE0020.severity = warning
+dotnet_diagnostic.IDE0038.severity = warning
 
-# Collection initialization can be simplified.
-dotnet_diagnostic.IDE0028.severity = warning
+# IDE0066 Use switch expression
+#csharp_style_prefer_switch_expression = true
+dotnet_diagnostic.IDE0066.severity = silent
 
-# Use coalesce expression (non-nullable types).
-dotnet_diagnostic.IDE0029.severity = warning
+# IDE0078 Use pattern matching
+#csharp_style_prefer_pattern_matching = true
+dotnet_diagnostic.IDE0078.severity = silent # Requires C# 9
 
-# Use coalesce expression (nullable types).
-dotnet_diagnostic.IDE0030.severity = warning
+# IDE0083 Use pattern matching ('not' operator)
+#csharp_style_prefer_not_pattern = true
+dotnet_diagnostic.IDE0083.severity = silent # Requires C# 9
 
-# Use null propagation.
-dotnet_diagnostic.IDE0031.severity = warning
+# IDE0170 Simplify property pattern
+#csharp_style_prefer_extended_property_pattern = true
+dotnet_diagnostic.IDE0170.severity = silent # Requires C# 10
 
-# Use explicitly provided tuple name.
-dotnet_diagnostic.IDE0033.severity = warning
+## Code block preferences
 
-# Simplify 'default' expression.
-dotnet_diagnostic.IDE0034.severity = warning
+# IDE0011 Add braces
+csharp_prefer_braces = when_multiline #true
+dotnet_diagnostic.IDE0011.severity = silent
 
-# Modifiers are not ordered.
-dotnet_diagnostic.IDE0036.severity = warning
+# IDE0063 Use simple 'using' statement
+#csharp_prefer_simple_using_statement = true
+dotnet_diagnostic.IDE0063.severity = silent # Requires C# 8
 
-# Use pattern matching to avoid 'is' check followed by a cast (without variable).
-dotnet_diagnostic.IDE0038.severity = warning
+## 'using' directive preferences
 
-# Use local function instead of lambda.
-dotnet_diagnostic.IDE0039.severity = warning
+# IDE0065 'using' directive placement
+#csharp_using_directive_placement = outside_namespace
+dotnet_diagnostic.IDE0065.severity = silent
 
-# Raise a warning on build when default access modifiers are explicitly specified.
-dotnet_diagnostic.IDE0040.severity = warning
+## File header preferences
 
-# Use 'is null' check.
-dotnet_diagnostic.IDE0041.severity = warning
+# IDE0073 Require file header
+#file_header_template = unset
+# This rule does not allow us to enforce our desired header, as it prefixes the header lines with // comments, meaning we can't apply a region.
+dotnet_diagnostic.IDE0073.severity = none
 
-# Deconstruct variable declaration.
-dotnet_diagnostic.IDE0042.severity = warning
+## Namespace naming preferences
 
-# Make field readonly.
-dotnet_diagnostic.IDE0044.severity = warning
+# IDE0130 Namespace does not match folder structure
+#dotnet_style_namespace_match_folder = true
+# This rule doesn't appear to work (never reports violations)
```

Comment:
It does work for me when I test it alone. Do we have any conflicting other rules? (I didn't test this whole file yet.)

Reply:
I can get a refactoring for this is VS, but no matter what I try I cannot get it to report IDE0130 as warning in either VS or a command-line build. I don't know if the dotnet SDK version matters, maybe something changed recently.

![image](https://user-images.githubusercontent.com/3399086/224475400-0845f3db-128e-480e-9739-8f3567e15168.png)

Reply:
This certainly works when set to `warning`: 
![image](https://user-images.githubusercontent.com/7704140/226034860-e5aa3f6c-325c-46bd-8179-58a6435ab447.png)

Reply:
Can you make a follow-up PR for this? I'd be quite interested in enabling this rule - but I'm suffering from it-doesn't-work-on-my-machine syndrome and I don't want to enable it without somebody confirming the local IDE and command-line experience is working as expected.

Reply:
Ok.